docs/ref/checks.txt

```diff
@@ -99,6 +99,12 @@ Some checks may be registered with multiple tags.
 Core system checks
 ==================
 
+Asynchronous
```

Comment:
I'm retitling this to 'asynchronous support' and adding an intro paragraph with links - it seems we are more explicit with the titling here